spec/index.bs

```diff
@@ -1970,6 +1981,55 @@ For more information on issuing CORS requests for image and video elements, cons
 - [[html#the-img-element]] <{img}>
 - [[html#media-elements]] {{HTMLMediaElement}}
 
+## Task Sources ## {#task-sources}
+
+### WebGPU Task Source ### {#-webgpu-task-source}
+
+WebGPU defines a new [=task source=] called the <dfn dfn>WebGPU task source</dfn>.
+It is used for the {{GPUDevice/uncapturederror}} event and {{GPUDevice}}.{{GPUDevice/lost}}.
+
+<div algorithm>
+    To <dfn abstract-op>queue a global task for {{GPUDevice}}</dfn> |device|,
+    with a series of steps |steps|:
+
+    1. [=Queue a global task=] on the [=WebGPU task source=], with the global object that was used
+        to create |device|, and the steps |steps|.
+</div>
+
+### Automatic Expiry Task Source ### {#-automatic-expiry-task-source}
+
+WebGPU defines a new [=task source=] called the <dfn dfn>automatic expiry task source</dfn>.
+It is used for the automatic, timed expiry (destruction) of certain objects:
+
+- {{GPUTexture}}s returned by {{GPUCanvasContext/getCurrentTexture()}}
+- Issue(gpuweb/gpuweb#3324): {{GPUExternalTexture}}s created from {{HTMLVideoElement}}s?
+
+<div algorithm>
+    To <dfn abstract-op>queue an automatic expiry task</dfn>
+    with {{GPUDevice}} |device| and a series of steps |steps|:
+
+    1. [=Queue a global task=] on the [=automatic expiry task source=], with the global object that
+        was used to create |device|, and the steps |steps|.
+</div>
+
+Tasks from the [=automatic expiry task source=] **should** be processed with high priority; in
+particular, once queued, they **should** run before user-defined (JavaScript) tasks.
+This results in stricter behavior that helps developers write more portable applications, but
+developers are still strongly encouraged to test in multiple implementations.
+
+<div class=note>
+    Note: Since task source priority is not normatively defined, there is a range of points at which
+    tasks from this task queue can execute. In practice, this means these tasks can be implemented
+    with fixed steps inside the [[HTML#event-loop-processing-model|HTML processing model]].
+    All of the following options are valid:
+
+    - Immediately after running any task.
+    - Immediately after running animation frame callbacks.
+    - Immediately before rendering.
+        (In the case of canvas texture expiry, this is already done in
+        [$update the rendering of the WebGPU canvas$], so the task is allowed to be omitted entirely.)
```

Comment:
Recognizing that this is non-normative, I still feel like this phrasing is slightly risky. One of the reasons we set off on this whole "When does it expire" journey is that we found cases in Chrome where the rendering wasn't being performed (tab was in the background, etc) and so the texture never expired. I'd hate to have other implementations look at this and say "Ah, we'll just expire on composite and skip the whole scheduling thing" and be bitten by the same bug.

I'm not sure exactly how to phrase it, but I think I'd want this to state more clearly something along the lines of "If the texture is expired in [$update the rendering of the WebGPU canvas$] then a separate expiry task is not necessary, but a task should be scheduled if the canvas rendering is not updated."?

Reply:
It's intended that that would be an allowed implementation. Wrote a warning into the note while trying not to say it's disallowed.